app/controllers/api/v1/labwares_controller.rb

```diff
@@ -0,0 +1,70 @@
+# See README.md for copyright details
+
+class Api::V1::LabwaresController < Api::V1::ApplicationController
+  before_action :set_labware, only: [:show, :create, :update]
+
+  # GET /labwares
+  def index
+    @labwares = Labware.all
+
+    render json: @labwares, include: included_relations_to_render
+  end
+
+  # GET /labwares/1
+  def show
+    render json: @labware, include: included_relations_to_render
+  end
+
+  # POST /labwares
+  def create
+    @labware = Labware.new(labware_params)
+
+    if @labware.save
+      render json: @labware, status: :created, include: included_relations_to_render
+    else
+      render json: @labware.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /labwares/1
+  def update
+    if @labware.update(labware_params)
+      render json: @labware, include: included_relations_to_render
+    else
+      render json: @labware.errors, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_labware
+    @labware = Labware.find_by(uuid: params[:id])
+  end
+
+  # Only allow a trusted parameter "white list" through.
+  def labware_params
+    params = (labware_json_params[:attributes] or {}).merge(uuid: labware_json_params[:id]).delete_if { |k, v| v.nil? }
+
+    if labware_json_params[:relationships] and
+        labware_json_params[:relationships][:labware_type] and
+        labware_json_params[:relationships][:labware_type][:data] and
+        labware_json_params[:relationships][:labware_type][:data][:attributes]
+      params = params.merge(labware_type: LabwareType.find_by(labware_json_params[:relationships][:labware_type][:data][:attributes]))
```

Comment:
This condition is very ugly.
It is always checking the existence of the labware type even when creating a labware.
We should find a way to refactor it